Library/PKPaymentRequest+Helpers.swift

```diff
@@ -84,13 +95,56 @@ extension PKPaymentRequest {
 
     paymentSummaryItems.append(
       PKPaymentSummaryItem(
-        label: postCampaignPledgingActive ? Strings.Kickstarter_payment_summary() : Strings
-          .Kickstarter_if_funded(),
+        label: Strings.Kickstarter_if_funded(),
         amount: total,
         type: .final
       )
     )
 
     return paymentSummaryItems
   }
+
+  private static func paymentSummaryItems(
+    for data: PostCampaignPaymentAuthorizationData
+  ) -> [PKPaymentSummaryItem] {
+    var paymentSummaryItems: [PKPaymentSummaryItem] = []
+
+    paymentSummaryItems.append(
+      PKPaymentSummaryItem(
+        label: data.hasNoReward ? Strings.Total() : Strings.activity_creator_reward(),
+        amount: NSDecimalNumber(value: data.subtotal),
+        type: .final
+      )
+    )
+
+    if data.bonus > 0 {
+      paymentSummaryItems.append(
+        PKPaymentSummaryItem(
+          label: Strings.Bonus(),
+          amount: NSDecimalNumber(value: data.bonus),
+          type: .final
+        )
+      )
+    }
+
+    if data.shipping > 0 {
+      paymentSummaryItems.append(
+        PKPaymentSummaryItem(
+          label: Strings.Shipping(),
+          amount: NSDecimalNumber(value: data.shipping),
+          type: .final
+        )
+      )
+    }
+
+    paymentSummaryItems.append(
+      PKPaymentSummaryItem(
+        label: Strings.Kickstarter_payment_summary(),
+        amount: NSDecimalNumber(value: data.total),
+        type: .final
+      )
+    )
+
+    return paymentSummaryItems
```

Comment:
This is all much simpler. The actual logic to calculate the subtotal happens in the view model, instead of here.